### PR TITLE
Add article about RSpec split by example in GitHub Actions

### PR DESCRIPTION
**Pull request by shadre**

### CODE REVIEW

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+title:  "Split your RSpec individual test examples between Github Action nodes with Knapsack Pro"
```
Comment (shadre) [inline _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:3]: WIP title - need a better idea

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
+          KNAPSACK_PRO_CI_NODE_INDEX: ${{ matrix.ci_node_index }}
+          KNAPSACK_PRO_FIXED_QUEUE_SPLIT: true
+          KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES: true
+          KNAPSACK_PRO_LOG_LEVEL: info
+        run: |
+          bundle exec rake knapsack_pro:queue:rspec
+{% endhighlight %}
+
+Let us know in the comments what you think about this solution. Also, don't hesitate to ask questions if you encounter any troubles with configuring GitHub Actions in your project - we'd be happy to help.
```
Comment (ArturT) [inline _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:126]: add link to https://knapsackpro.com/faq/question/how-to-split-slow-rspec-test-files-by-test-examples-by-individual-it

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+title:  "Split your RSpec individual test examples between Github Action nodes with Knapsack Pro"
+date:   2020-06-13 12:00:00 +0100
+author: "Knapsack Pro"
+categories: techtips continuous_integration
+og_image: "/images/blog/posts/example-article/image.jpg"
```
Comment (ArturT) [inline _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:7]: for myself: do image here

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md (hunk 1 above)
Comment (ArturT) [reply _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:3]: The idea for the title with some keywords and keeping Github Actions in the title name closer to the beginning of the title:

* How to run slow RSpec files on Github Actions with parallel jobs by doing an auto split of the spec file by test examples

If we change the title we need to update MD file name for this blog post.

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+Second, we belive that the code organization should be based on other considerations. How you create your files and classes is most likely a result of following some approach agreed upon in your project. Dividing classes into smaller ones so that the CI build can run faster encroaches on your conventions. It might be more disturbing to some than the others, but we feel it's fair to say it'd be best to avoid - if there was a better way to achieve the same...
+
+## Enter split by test examples
+
+As you certainly know, RSpec allows us to run individual examples instead of whole files. We decided to take advantage of that, and solve the problem of bottleneck test files by gathering information about individual examples from such slower files. Such examples are then dynamically distributed between your parallel nodes and run individually, so no individual file can be a bottleneck for the whole build. What's important, no additional work is needed - this is done automatically by the gem. Each example is run in its correct context that's set-up exactly the same as if you had run the whole file.
```
Comment (ArturT) [inline _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:28]: ` by the gem.` => ` by the knapsack_pro gem`. It wasn't obvious what kind of gem you meant.

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          RAILS_ENV: test
+        run: |
+          bin/rails db:prepare
+      - name: Run tests
+        env:
+          PGHOST: localhost
+          PGUSER: postgres
+          RAILS_ENV: test
+          KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC: ${{ secrets.KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC }}
+          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
```
Comment (ArturT) [inline _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:117]: signs `${{ something }}` are not rendered properly. We need to use `raw` see:

```
{% highlight yml %}
{% raw %}
code
{% endraw %}
{% endhighlight %}
```

This was done in _posts/2020-03-05-how-to-merge-cypress-test-reports-generated-by-mochawesome-on-github-actions.md line 44

File: _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md (hunk 2 above)
Comment (ArturT) [reply _posts/2020-06-13-split-your-rspec-individual-test-examples-between-github-action-nodes.md:126]: maybe we could also add here my video from above link just to encourage someone to see live example. More details can be found in the article https://knapsackpro.com/faq/question/how-to-split-slow-rspec-test-files-by-test-examples-by-individual-it 

Would be cool if link to article https://knapsackpro.com/faq/question/how-to-split-slow-rspec-test-files-by-test-examples-by-individual-it is attached to good keywords like `How split of slow RSpec test files works` or something else where we mention keywords like RSpec

**Pull request merged**